### DISCUSSION TITLE
Wie kann man zum "Grid Profile" beitragen?

### DISCUSSION THREAD

**Discussion by NintendoTito** [tbnobody/OpenDTU#1394] [Q&A]
Hallo zusammen, ich habe die neue Funktion zum Grid Profile gesehen. Der Link zu der entsprechenden Seite ist für mich aber nicht so "selbsterklärend"... 😬

Kann mir das hier jemand bißchen genauer erklären wie man da ggf. helfen kann?

**Comment by DPO99** [comment]
Nunja du kannst prüfen, ob es dein HM Model und dein Grid Profil da schon gibt, dann gibts m.e. nichts zu tun. Wenn du einen Hm oder ein Grid Profil hast, dass es dort noch nicht gibt, dann einen Eintrag erstellen.

**Reply by NintendoTito** [reply]
Ich sehe bei mir nur ne alphanummerische Matrix und darüber steht "Unterstütze die Entwicklung". Woran sehe ich jetzt ob mein Grid Profile schon bekannt ist? Handelt es sich bei der Matrix ums das Grid Profile? Ist das spezifisch je Netzbetreiber?

**Reply by avandorp** [reply]
Es handelt sich um die hexadezimale Darstellung der Grid-Profile-Binärblobs wie Hoymiles sie verwendet. Wenn man dazu die Klartext-Info hat, kann man versuchen, das Format der Grid Profiles zu rekonstruieren um selber welche zu erstellen und ggf. auch hochzuladen. Keine Ahnung, ob das der Plan ist. Am Besten wäre wohl, einen der Cloud-Zugänge zu haben, bei welchen man sich beliebige Grid Profiles mit eigenen Parametern erstellen kann, dann könnte man das ganze am einfachsten rekonstruieren.

**Comment by noone2k** [comment]
grid profile parser in python
https://gist.github.com/noone2k/0b3a116a6f35286abef7199b62a0777a

**Reply by svalcken** [reply]
Wo kann ich das Grid Profil vollständig auslesen?

Ich habe nur in der OpenDTU 6 Hexadezimale Bytes gesehen.
Die Grid Profile hier sind aber erheblich länger.

Und nach dem Parser ist das aus der OpenDTU nicht das aktuell eingestellte Grid Profil.
Das hatte ich per S-Cloud auf DE VDE... eingestellt. Aber weder Modell noch Grid werden gerade von der DTU richtig erkannt.

**Reply by tbnobody** [reply]
The bytes shown in the inverter dialog are the complete response from the inverter. you could doublecheck the serial console after a restart of the dtu. Look for the request `GridOnProFilePara` and it's response.
